ext/include/opentelemetry/ext/http/client/curl/http_operation_curl.h

```diff
@@ -470,7 +477,47 @@ class HttpOperation
     }
 
     /* select() returns the number of signalled sockets or -1 */
-    res = select((int)sockfd + 1, &infd, &outfd, &errfd, &tv);
+    if (select((int)sockfd + 1, &infd, &outfd, &errfd, &tv) > 0)
+    {
+      if (for_recv)
+      {
+        res = FD_ISSET(sockfd, &infd);
```

Comment:
Some compiler will warning this(implicit convert from int to bool).We should use `res = 0 != FD_ISSET(sockfd, &infd);` here.